modules/bcftools/concat/main.nf

```diff
@@ -0,0 +1,39 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process BCFTOOLS_CONCAT {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::bcftools=1.11" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bcftools:1.11--h7c999a4_0"
+    } else {
+        container "quay.io/biocontainers/bcftools:1.11--h7c999a4_0"
+    }
+
+    input:
+    tuple val(meta), path(vcfs)
+
+    output:
+    tuple val(meta), path("*.gz"), emit: vcf
+    path  "*.version.txt"        , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    prefix       = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    bcftools concat \\
+        --output ${prefix}.vcf.gz \\
```

Comment:
Just to check, as bcftools isn't clear: does bcftools pick up the extension automatically and compress accordingly?

I see there is an `--output-type` flag which may be requried... (not used it myself ;) )

Reply:
Good idea, though I think `bcftools` does pick up the extension.

The `--output-type` could always be passed through the `options.args`, no?

Reply:
If it does so automatically then yes, fine to leave it to `options.args`. just wanted to double check 👍